code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -30,8 +30,10 @@
 /obj/item/borg/upgrade/reset/action(mob/living/silicon/robot/R)
 	if(..())
 		return
-
 	R.reset_module()
+	R.shown_robot_modules = 0
+	R.client.screen -= R.robot_modules_background
+	R.client.screen -= R.hud_used.module_store_icon
```

Comment:
These three lines should probably go into the `reset_module()` proc now that I think about it, just so we can call that proc from anywhere and still have the store icon removed.

Reply:
Put these into the `reset_module()` proc.